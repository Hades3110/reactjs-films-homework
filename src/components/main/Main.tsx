import FilmsFilter from './filmsFilter/FilmsFilter'
import FilmsList from './filmsList/FilmsList'
import styles from './main.module.scss'

const Main = () => (
    <main className={styles.main}>
        <FilmsFilter />
        <FilmsList />
    </main>
)

export default Main
