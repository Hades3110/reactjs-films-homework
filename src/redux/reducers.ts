import { combineReducers } from "redux";
import filmListPageCountReducer from "./filmListCountAndCategories/reducer";
import filmListSortReducer from "./filmListSort/reducer";
import mainBannerReducer from "./mainBanner/reducer";

const rootReducer = combineReducers({
    pageCounter: filmListPageCountReducer,
    sortType: filmListSortReducer,
    mainBanner: mainBannerReducer
})

export default rootReducer
