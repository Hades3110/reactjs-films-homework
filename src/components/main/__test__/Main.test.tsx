import { create } from "react-test-renderer";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Main from '../Main';
import rootReducer from '../../../redux/reducers';

describe('Main Component', () => {

  const store = createStore(rootReducer)

  it('Main snapshot test', () => {
    const component = create(<Provider store={store}><Main /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});