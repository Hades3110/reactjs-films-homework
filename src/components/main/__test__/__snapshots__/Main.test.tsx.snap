// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component Main snapshot test 1`] = `
<main
  className="main"
>
  <div
    className="filmFilter"
  >
    <div
      className="filmFilter__filter"
    >
      <button
        className="filmFilter__filter__btns activeCategories"
        onClick={[Function]}
      >
        Trending
      </button>
      <button
        className="filmFilter__filter__btns"
        onClick={[Function]}
      >
        Top Rated
      </button>
      <button
        className="filmFilter__filter__btns"
        onClick={[Function]}
      >
        Coming Soon
      </button>
    </div>
    <div
      className="filmFilter__arrange"
    >
      <button
        className="filmFilter__arrange__row activeSort"
        onClick={[Function]}
      >
        <div
          className="filmFilter__arrange__row__el"
        />
        <div
          className="filmFilter__arrange__row__el"
        />
        <div
          className="filmFilter__arrange__row__el"
        />
        <div
          className="filmFilter__arrange__row__el"
        />
      </button>
      <button
        className="filmFilter__arrange__oneColumn"
        onClick={[Function]}
      >
        <div
          className="filmFilter__arrange__oneColumn__el"
        />
        <div
          className="filmFilter__arrange__oneColumn__el"
        />
      </button>
    </div>
  </div>
  <img
    className="spinner"
    src="test-file-stub"
  />
</main>
`;
